src/utils/html.js

```diff
@@ -0,0 +1,38 @@
+export function injectCSPMetaTagIntoDataURI(dataURI) {
+  // data URI -> HTML
+  const prefix = 'data:text/html;base64,'
+  const base64 = dataURI.replace(prefix, '')
+  const html = atob(base64)
+
+  // inject CSP meta tag
+  const safeHTML = injectCSPMetaTagIntoHTML(html)
+
+  // HTML -> data URI
+  return `${prefix}${btoa(safeHTML)}`
+}
+
+export function injectCSPMetaTagIntoBuffer(buffer) {
+  // buffer -> HTML
+  const html = new TextDecoder().decode(buffer)
+
+  // inject CSP meta tag
+  const safeHTML = injectCSPMetaTagIntoHTML(html)
+
+  // HTML -> buffer
+  return new TextEncoder().encode(safeHTML)
+}
+
+export function injectCSPMetaTagIntoHTML(html) {
+  // HTML -> doc
+  const parser = new DOMParser()
+  const doc = parser.parseFromString(html, 'text/html')
+
+  // inject CSP meta tag
+  const meta = document.createElement('meta')
+  meta.httpEquiv = 'Content-Security-Policy'
+  meta.content = `default-src 'self'; script-src 'unsafe-inline'; style-src 'unsafe-inline'`
+  doc.head.appendChild(meta)
```

Comment:
Maybe also check if the html already has `Content-Security-Policy` and delete it?
What happens if there are two?

Reply:
Good call, thanks! Fixed here: https://github.com/hicetnunc2000/hicetnunc/pull/189/commits/2ce5c0ec337f384397b9ac15f6b6456551d70ab1